main.cpp

```diff
@@ -275,296 +300,281 @@ struct Bakery
+    bool selfResonate(float fResonance, double fInputGain);
 };
 
-/*
-Thing 7) envelope
-5 properties:
-    1) attack time (float)
-    2) decay time (float)
-    3) sustain level (double)
-    4) release time (float)
-    5) curve (float)
-3 things it can do:
-    1) create long sustained signals
-    2) create short staccato signals
-    3) loop over time
- */
+void Filter::cutLows(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting low frequencies below: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+void Filter::cutHighs(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting high frequencies above: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+bool Filter::selfResonate(float fResonance, double fInputGain)
+{
+    return fResonance > 0.9f && fInputGain > -35.0;
+}
 
 struct Envelope
 {
-    //1) attack time (float)
     float attackTime = 22.5f;
-    //2) decay time (float)
     float decayTime = 330.2f;
-    //3) sustain level (double)
     double sustainLevel = 0.99228;
-    //4) release time (float)
     float releaseTime = 675.1f;
-    //5) curve (float)
     float curve = 0.9f;
 
-    //1) create long sustained signals
-    void padPreset(float curve);
-    //2) create short staccato signals
-    void staccatoPreset(float curve);
-    //3) loop over time
-    bool loopOver(float attackTime, float decayTime, float releaseTime);
+    void padPreset(float eCurve);
+    void staccatoPreset(float eCurve);
+    bool loopOver(float eAttackTime, float eDecayTime, float eReleaseTime);
 };
 
-/*
-Thing 8) lfo
-5 properties:
-    1) fade-in time (float)
-    2) frequency (double)
-    3) shape (int)
-    4) retrigger behaviour (int)
-    5) mono / poly switch (bool)
-3 things it can do:
-    1) reset
-    2) modulate
-    3) change shape
- */
+void Envelope::padPreset(float eCurve)
+{
+    std::cout << "Setting envelope to slow setting, with curve of: " << eCurve << std::endl;
+}
+void Envelope::staccatoPreset(float eCurve)
+{
+    std::cout << "Setting envelope to fast setting, with curve of: " << eCurve << std::endl;
+}
+bool Envelope::loopOver(float eAttackTime, float eDecayTime, float eReleaseTime)
+{
+    bool eSwitch = false;
+
+    if ((eAttackTime + eDecayTime + eReleaseTime) < 700.f)
+    {
+        eSwitch = true;
+        return true;
+    }
+
+    return false;
+}
 
 struct LFO
 {
-    //1) fade-in time (float)
     float fadeIn = 45.3f;
-    //2) frequency (double)
     double freq = 0.074;
-    //3) shape (int)
     int shape = 2;
-    //4) retrigger behaviour (int)
     int retrig = 3;
-    //5) mono / poly switch (bool)
     bool polyMode = false;
 
-    //1) reset
     void resetState();
-    //2) modulate
-    void modulateFreq(double freq, float modFreq, float intensity);
-    //3) change shape
-    void changeSpeed(int shape);
+    void modulateFreq(double lFreq, float modFreq, float intensity);
+    void changeSpeed(int lShape);
```

Comment:
Review our Slack discussion of using member variables in member functions directly. 
Change your argument list so that your function arguments are not duplicates of your member variables.
This can mean changing the function name (meaning the function purpose has changed) 
or it could mean changing the function arguments.
Remember, name things what they are or what they do.
- [ ] revise